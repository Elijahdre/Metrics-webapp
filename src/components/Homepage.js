import { useEffect } from 'react';
import { useDispatch } from 'react-redux';

import { fetchCrypto } from '../redux/crypto/crypto';
import Currencies from './Currencies';

function Homepage() {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCrypto());
  }, []);

  return (
    <div>
      <Currencies />
    </div>
  );
}

export default Homepage;
