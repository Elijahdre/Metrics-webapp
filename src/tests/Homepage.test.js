import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';

import Homepage from '../components/Homepage';

describe('Homepage', () => {
  it('renders correctly', () => {
    const tree = Renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Homepage />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});