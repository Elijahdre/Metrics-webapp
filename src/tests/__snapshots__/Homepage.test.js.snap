// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage renders correctly 1`] = `
<div>
  <main>
    <section
      className="main-container"
    >
      <img
        alt="crypto wallet"
        src="digital-wallet.png"
      />
      <div>
        <h1>
          The Worldâ€™s Fastest Growing
          <br />
          Crypto App
        </h1>
      </div>
    </section>
    <section
      className="currency-data"
    >
      <div
        className="card"
      >
        <a
          href="/details/:Bitcoin"
          onClick={[Function]}
        >
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            />
          </svg>
        </a>
        <div
          className="image"
        >
          <img
            alt="Bitcoin"
            src="bitcoin.png"
          />
        </div>
        <h2>
          Bitcoin
        </h2>
      </div>
      <div
        className="card"
      >
        <a
          href="/details/:Ethereum"
          onClick={[Function]}
        >
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            />
          </svg>
        </a>
        <div
          className="image"
        >
          <img
            alt="Ethereum"
            src="ethereum.png"
          />
        </div>
        <h2>
          Ethereum
        </h2>
      </div>
      <div
        className="card"
      >
        <a
          href="/details/:Tether"
          onClick={[Function]}
        >
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            />
          </svg>
        </a>
        <div
          className="image"
        >
          <img
            alt="Tether"
            src="tether.png"
          />
        </div>
        <h2>
          Tether
        </h2>
      </div>
      <div
        className="card"
      >
        <a
          href="/details/:USD Coin"
          onClick={[Function]}
        >
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            />
          </svg>
        </a>
        <div
          className="image"
        >
          <img
            alt="USD Coin"
            src="usd.png"
          />
        </div>
        <h2>
          USD Coin
        </h2>
      </div>
      <div
        className="card"
      >
        <a
          href="/details/:BNB"
          onClick={[Function]}
        >
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            />
          </svg>
        </a>
        <div
          className="image"
        >
          <img
            alt="BNB"
            src="binance.png"
          />
        </div>
        <h2>
          Binance coin
        </h2>
      </div>
      <div
        className="card"
      >
        <a
          href="/details/:XRP"
          onClick={[Function]}
        >
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            />
          </svg>
        </a>
        <div
          className="image"
        >
          <img
            alt="XRP"
            src="xrp.png"
          />
        </div>
        <h2>
          XRP
        </h2>
      </div>
      <div
        className="card"
      >
        <a
          href="/details/:Cardano"
          onClick={[Function]}
        >
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            />
          </svg>
        </a>
        <div
          className="image"
        >
          <img
            alt="Cardano"
            src="cardano.png"
          />
        </div>
        <h2>
          Cardano
        </h2>
      </div>
    </section>
  </main>
</div>
`;
